Source/RapyutaSimulationPlugins/Private/Robots/RRRobotROS2Interface.cpp

```diff
@@ -441,33 +446,37 @@ void URRRobotROS2Interface::JointStateCallback(const UROS2GenericMsg* Msg)
     }
 }
 
-
 URRRobotROS2InterfaceComponent::URRRobotROS2InterfaceComponent()
 {
     ROS2Interface = CastChecked<URRRobotROS2Interface>(
         URRUObjectUtils::CreateSelfSubobject(this, ROS2InterfaceClass, FString::Printf(TEXT("%sROS2Interface"), *GetName())));
 
-    if ( nullptr == Robot )
+    if (nullptr == Robot)
     {
-        UE_LOG_WITH_INFO(LogTemp, Warning, TEXT("[URRRobotROS2InterfaceComponent][URRRobotROS2InterfaceComponent] Robot is nullptr. Trying to get owner as robot."));
+        UE_LOG_WITH_INFO(LogTemp,
+                         Warning,
+                         TEXT("[URRRobotROS2InterfaceComponent][URRRobotROS2InterfaceComponent] Robot is nullptr. Trying to get "
+                              "owner as robot."));
         Robot = Cast<ARRBaseRobot>(GetOwner());
     }
 
-    if ( nullptr != Robot )
+    if (nullptr != Robot)

```

Comment:
@yuokamoto, just to be curious, any difference when we check for `Robot` for `ROS2Interface->ROSSpawnParameters = Robot->ROSSpawnParameters;` inside `URRRobotROS2InterfaceComponent::OnComponentCreated()`, when `GetOwner()` may give non-nullptr, versus ctor when it could be too early to be assigned?

Reply:
If I remember correctly, I haven't tried OnComponentCreated.
What I struggle was to make it possible to change param from bp as much as poosible. it maybe possible to use OnComponentCreated but honestly says, there are too many those method to decide which one is better.